src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -81,7 +86,80 @@ protected bool InitializeFileWatcherIfEnabled()
             return false;
         }
 
-        public abstract Task Invoke(object[] parameters);
+        public async Task Invoke(object[] parameters)
+        {
+            FunctionStartedEvent startedEvent = null;
+
+            // We require the ExecutionContext, so this will throw if one is not found.
+            ExecutionContext functionExecutionContext = (ExecutionContext)parameters.First(p => p?.GetType() == typeof(ExecutionContext));
```

Comment:
changed to `parameters.OfType<ExecutionContext>().First()`

---
In reply to: [77682305](https://github.com/Azure/azure-webjobs-sdk-script/pull/604#discussion_r77682305) [](ancestors = 77682305)